packages/@uppy/companion-client/src/RequestClient.js

```diff
@@ -167,9 +168,24 @@ export default class RequestClient {
     }
   }
 
-  async get (path, skipPostResponse) { return this.#request({ path, skipPostResponse }) }
+  async get (path, options = undefined) {
+    // TODO: remove boolean support for options that was added for backward compatibility.
+    // eslint-disable-next-line no-param-reassign
+    if (typeof options === 'boolean') options = { skipPostResponse: options }
+    return this.#request({ ...options, path })
+  }
 
-  async post (path, data, skipPostResponse) { return this.#request({ path, method: 'POST', data, skipPostResponse }) }
+  async post (path, data, options = undefined) {
```

Comment:
I think data should be part of options because you might want to post/put/delete without any data too?

Reply:
nevermind it's because of backward compatibility, right?

Reply:
For `POST`  it would make very little sense, no, for `DELETE` I could see it happening, but since it would be a breaking change, I think it's just easier to pass `undefined` for `data` when that happens.

Reply:
How do you know if `data` is the data or the options? Like that `client.delete('/endpoint', { data: { partNumber: 1 } }`, should it be interpreted as is or as `client.delete('/endpoint', { data: { data: { partNumber: 1 } } }`?